crates/sui-types/src/messages.rs

```diff
@@ -460,11 +643,56 @@ impl Command {
                     call.function.as_str(),
                 ));
                 fp_ensure!(!is_blocked, SuiError::BlockedMoveFunction);
+                let mut type_arguments_count = 0;
+                for tag in call.type_arguments.iter() {
+                    type_arguments_count +=
+                        type_tag_validity_check(tag, config, 1, type_arguments_count)?;
+                    fp_ensure!(
+                        type_arguments_count < config.max_type_arguments() as usize,
+                        SuiError::SizeLimitExceeded {
+                            limit: "maximum type arguments in a call transaction".to_string(),
+                            value: config.max_arguments().to_string()
+                        }
+                    );
+                }
+                fp_ensure!(
+                    call.arguments.len() < config.max_arguments() as usize,
+                    SuiError::SizeLimitExceeded {
+                        limit: "maximum arguments in a move call".to_string(),
+                        value: config.max_arguments().to_string()
+                    }
+                );
+            }
+            Command::TransferObjects(v, _) => {
+                fp_ensure!(
+                    v.len() < config.max_arguments() as usize,
```

Comment:
I wonder if we should conflate `max_argument` here with the move call arguments... 
fine leaving it that way fwiw, we can change it later